cuqipy_fenics/testproblem.py

```diff
@@ -147,3 +155,275 @@ def form(m,u,p):
+        Standard deviation of the noise relative to the exact data. By default, the noise is 1% (=0.01) of the exact data.
+
+    field_type : str, Default None
+        | Field type of the forward model domain. The accepted values are:
+        | "KL": a :class:`MaternExpansion` geometry object will be created and set as a domain geometry.
+        | None: a :class:`FEniCSContinuous` geometry object will be created and set as a domain geometry.
+
+    field_params : dict, Default None
+        | A dictionary of keyword arguments that the underlying geometry accepts. (Passed to the underlying geometry when field type is "KL" or None). For example, for "KL" field type, the dictionary can be `{"length_scale": 0.1, "num_terms": 32}`. If None is passed as field_type, this argument is ignored.
+
+    mapping : str or callable , Default None
+        | mapping to parametrize the Bayesian parameters. If None, no mapping is applied. If provided as callable, it should take a FEniCS function (of the unknown parameter) as input and return a FEniCS form, e.g. `lambda m: ufl.exp(m)`.
+        If provided as string, it can take one of the values: 
+        | 'exponential' : Parameterization in which the unknown parameter becomes the log of the diffusion coefficient.
+
+    prior : cuqi.distribution.Distribution, Default Gaussian
+        | Distribution of the prior. Needs to be i.i.d standard Gaussian if field_type is "KL". The prior name property, i.e., `prior.name` is expected to be "x".
+    """
+
+    def __init__(self, dim=None, bc_types=None, bc_values=None,
+                 exactSolution=None, f=None, relative_noise_std=None, field_type=None,
+                 field_params=None, mapping=None, prior=None):
+
+        # Create the mesh
+        if dim is None:
+            dim = (32, 32)
+        mesh = dl.UnitSquareMesh(dim[0], dim[1])
+
+        # Create the function space
+        V = dl.FunctionSpace(mesh, 'Lagrange', 1)
+
+        # Set up boundary conditions
+        if bc_types is None:
+            bc_types = ['Dirichlet', 'Dirichlet', 'Dirichlet', 'Dirichlet']
+        elif len(bc_types) != 4:
+            raise ValueError(
+                "The length of bc_types list should be 4. The list should be ordered as follows: [left, bottom, right, top]")
+        elif all(bc_type.lower() in ['neumann'] for bc_type in bc_types):
+            raise ValueError(
+                "All boundary conditions cannot be Neumann. At least one boundary condition should be Dirichlet.")
+        if bc_values is None:
+            bc_values = [0, 0, 0, 0]
+        elif len(bc_values) != 4:
+            raise ValueError(
+                "The length of bc_values list should be 4. The list should be ordered as follows: [left, bottom, right, top]")
+        bc_values = [to_dolfin_expression(bc_value) for bc_value in bc_values]
+
+        subdomains = self._create_boundaries_subdomains()
+        dirichlet_bcs = self._set_up_dirichlet_bcs(
```

Comment:
I'm unsure what's the best naming convention here, I see the reason to make it "bcs" for plural of "bc", however it is harder to understand, at least at first, since one may not realize it is for plural but think it is part of the abbreviation "bcs". I guess "BCs" would be clearer, but not sure if it fits with convention elsewhere. Could also keep it simply "bc" implicitly covering plural boundaries conditions, but might be less clear.